### PR TITLE
MGMT-5302: Removing deprecated /host_requirements endpoint and related…

### PR DESCRIPTION
**Pull request by pkliczewski**
# Assisted Pull Request

## Description

This PR removes `/host_requirements` endpoint and related model.

Signed-off-by: Piotr Kliczewski <piotr.kliczewski@redhat.com>

## List all the issues related to this PR

- [ ] New Feature
- [ ] Bug fix
- [ ] Tests
- [ ] Documentation
- [ ] CI/CD 

## What environments does this code impact?

- [ ] Automation (CI, tools, etc)
- [ ] Cloud
- [ ] Operator Managed Deployments
- [x] None

## How was this code tested?

- [ ] assisted-test-infra environment
- [ ] dev-scripts environment
- [ ] Reviewer's test appreciated
- [ ] Waiting for CI to do a full test run
- [ ] Manual (Elaborate on how it was tested)
- [x] No tests needed

## Assignees

/cc @
/cc @

## Checklist

- [ ] Title and description added to both, commit and PR.
- [ ] Relevant issues have been associated (see [CONTRIBUTING] guide)
- [ ] Reviewers have been listed
- [ ] This change does not require a documentation update (docstring, `docs`, README, etc)
- [ ] Does this change include unit-tests (note that code changes require unit-tests)

## Reviewers Checklist

- [ ] Are the title and description (in both PR and commit) meaningful and clear?
- [ ] Is there a bug required (and linked) for this change?
- [ ] Should this PR be backported?

[Kubernetes community documentation]: https://github.com/kubernetes/community/blob/master/contributors/guide/pull-requests.md#commit-message-guidelines
[CONTRIBUTING]: https://github.com/openshift/assisted-service/blob/master/CONTRIBUTING.md

### CODE REVIEW

**Comment by pkliczewski** [comment]
@avishayt @flaper87 @YuviGold @jakub-dzon please review

File: internal/bminventory/inventory.go [hunk 1]
```diff
@@ -4521,38 +4520,6 @@ func (b *bareMetalInventory) GetPreflightRequirements(ctx context.Context, param
 	return installer.NewGetPreflightRequirementsOK().WithPayload(requirements)
 }
 
-func (b *bareMetalInventory) GetHostRequirements(_ context.Context, params installer.GetHostRequirementsParams) middleware.Responder {
-	requirements, err := b.hwValidator.GetDefaultVersionRequirements()
```
Comment (YuviGold) [inline internal/bminventory/inventory.go:4525]: Could `GetDefaultVersionRequirements` be removed as well? I don't see it being used.

**Review by filanov — APPROVED** [review]

File: internal/bminventory/inventory.go (hunk 1 above)
Comment (pkliczewski) [reply internal/bminventory/inventory.go:4525]: Sure

Comment (pkliczewski) [reply internal/bminventory/inventory.go:4530]: Removed

**Pull request merged**